Objects/typevarobject.c

```diff
@@ -364,26 +364,24 @@ typevar_new_impl(PyTypeObject *type, PyObject *name, PyObject *constraints,
         }
     }
 
-    if (constraints != NULL) {
-        if (!PyTuple_CheckExact(constraints)) {
-            PyErr_SetString(PyExc_TypeError,
-                            "constraints must be a tuple");
-            return NULL;
-        }
-        Py_ssize_t n_constraints = PyTuple_GET_SIZE(constraints);
-        if (n_constraints == 1) {
-            PyErr_SetString(PyExc_TypeError,
-                            "A single constraint is not allowed");
-            Py_XDECREF(bound);
-            return NULL;
-        } else if (n_constraints == 0) {
-            constraints = NULL;
-        } else if (bound != NULL) {
-            PyErr_SetString(PyExc_TypeError,
-                            "Constraints cannot be combined with bound=...");
-            Py_XDECREF(bound);
-            return NULL;
-        }
+    if (!PyTuple_CheckExact(constraints)) {
```

Comment:
Maybe we can actually turn this into an assert? If I understand correctly, AC should guarantee it's always a tuple.